spec/controllers/catalog_controller_spec.rb

```diff
@@ -32,5 +32,47 @@
         expect(controller).to have_search_parameters
       end
     end
+
+    context "when not logged in" do
```

Comment:
Nice. Thanks, and... I agree with all of the above. I think there is a lot of value in having things wired up and driving through the real ability model here. Whether the permissions are best validated at the controller is TBD in my mind (though I suspect not).

I'm interested to think more about this down the line. I find that every time there is a binding to `current_user` or things like params/session outside of the "request scope" (controller/view), I regret it. We might get some mileage out of testing the user/group/object permission application outside of a controller and then leaning on a verifying double of the persister to validate the controller behavior. It's not so much a speed thing as a "this test was a pain to set up and could break for three different reasons" thing.

Just thoughts for later. Happy to have this go ahead as-is.